nion/instrumentation/Acquisition.py

```diff
@@ -0,0 +1,1157 @@
+                    next_source_index = source_index + count
+                    new_source_data = old_source_data[source_index:next_source_index].reshape((1, count) + old_source_data.shape[1:])
+                    new_state = DataStreamStateEnum.PARTIAL  # always partial, otherwise would have been sent in previous section
+                    self.data_available_event.fire(DataStreamEventArgs(self, channel, new_data_metadata, new_source_data, None, new_source_slice, new_state))
+                    # source_index = next_source_index  # no need for this
+                    index += count
+                    count -= count
+                    assert count is not None  # for type checker bug
+                assert count == 0  # everything has been accounted for
+                self.__indexes[channel] = next_index % collection_count
+                self.__sub_slice_indexes[channel] = next_sub_slice_index
+                self.__needs_starts[channel] = next_sub_slice_index == collection_sub_slice_length
+            if new_state == DataStreamStateEnum.COMPLETE:
+                self._sequence_next(channel)
+        else:
+            # incoming data is partial
+            # form the new slice with indexes of 0 for each collection dimension and the incoming slice for the
+            # remaining dimensions. add dimensions for collection dimensions to the new source data.
+            new_source_slice = (index_slice(0), ) * collection_rank + tuple(data_stream_event.source_slice)
+            new_source_data = data_stream_event.source_data.reshape((1,) * collection_rank + data_stream_event.source_data.shape)
+            # new state is complete if data chunk is complete and it will put us at the end of our collection.
+            new_state = DataStreamStateEnum.PARTIAL
+            next_index = index
+            next_sub_slice_index = sub_slice_index
+            if data_stream_event.state == DataStreamStateEnum.COMPLETE:
+                next_index += 1
+                next_sub_slice_index += 1
+                if next_index == collection_count:
+                    new_state = DataStreamStateEnum.COMPLETE
+            self.data_available_event.fire(DataStreamEventArgs(self, channel, new_data_metadata, new_source_data, None, new_source_slice, new_state))
+            self.__indexes[channel] = next_index % collection_count
+            self.__sub_slice_indexes[channel] = next_sub_slice_index
+            self.__needs_starts[channel] = next_sub_slice_index == collection_sub_slice_length
+            if new_state == DataStreamStateEnum.COMPLETE:
+                self._sequence_next(channel)
+
+
+class SequenceDataStream(CollectedDataStream):
+    """Collect a data stream into a sequence of datums.
+
+    This is a subclass of CollectedDataStream.
+    """
+    def __init__(self, data_stream: DataStream, count: int, calibration: typing.Optional[Calibration.Calibration] = None):
+        super().__init__(data_stream, (count,), (calibration,))
```

Comment:
Using the plain default (`None`) for `calibration` here is problematic because it will be passed on to the base class `__init__` method where is is treated as a valid calibration. So the resulting data stream will have a `None` calibration as the calibration for the sequence axis, which causes errors when you are for example trying to put the data into a data item. Since `None` is not handled by the base class, it should be handled here and an empty calibration should be passed along.